src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/ContactsController.php

```diff
@@ -105,43 +110,92 @@ public function searchAction(Request $request)
     /**
      * Display the Contact creation form.
      *
-     * @AdminSecurity("is_granted('create', request.get('_legacy_controller'))", message="You do not have permission to add this.")
+     * @AdminSecurity(
+     *     "is_granted('create', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_contacts_index",
+     *     message="You do not have permission to add this."
+     * )
      *
-     * @return RedirectResponse
+     * @param Request $request
+     *
+     * @return Response
      */
-    public function createAction()
+    public function createAction(Request $request)
     {
-        $legacyLink = $this->getAdminLink('AdminContacts', [
-            'addcontact' => 1,
-        ]);
+        $contactFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.contact_form_builder');
+        $contactForm = $contactFormBuilder->getForm();
+        $contactForm->handleRequest($request);
+
+        try {
+            $contactFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.contact_form_handler');
+            $result = $contactFormHandler->handle($contactForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
```

Comment:
you should check for `getIdentifiableObjectId()` instead, like you did in `editAction`.